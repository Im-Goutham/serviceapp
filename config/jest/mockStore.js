export const initialState = {
    user: {
      user: {},
      backButton: false
    },
    photo: {
      photoView: false,
      images: [],
      index: 0
    }
  }