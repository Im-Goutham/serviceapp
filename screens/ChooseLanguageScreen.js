import React, { Component } from 'react';
import {
    View,
    StyleSheet,
    ScrollView,
    Image,
    TouchableOpacity,
    Dimensions,
    Platform,
    Text, TouchableHighlight
} from 'react-native';
 import {  Icon } from 'native-base'
import Advertisement from '../components/Advertisement';
import LanguageList from '../components/LanguageList';
import Map from '../components/Map';
import Header from '../components/Header';
import Modal from 'react-native-modalbox';
import LinearGradient from 'react-native-linear-gradient';
// import Icon from 'react-native-vector-icons/EvilIcons';
import {scale} from '../global';

import HeaderScreen from './HeaderScreen';

var {height, width} = Dimensions.get('window');
let tabItems = ["List View", "Map View"];

let logo = require('../images/logo.png');
let menu = require('../assets/icons/menu.png');
let back_arrow = require('../assets/icons/arrow_left.png');
let border_img = require('../images/border_img.png');

let maplocations = {
    data : [
         {
      jobtitle: 'Need Cook',
      icon: require('../assets/icons/crown.png'),
      image: require('../images/cook.png'),
      detail: "Lorem Ipsum has been the industrys standard dummy text ever",
    },
    {
      jobtitle: 'Need Carpenter',
      icon: require('../assets/icons/crown.png'),
      image: require('../images/tutorial.png'),
      detail: "Lorem Ipsum has been the industrys standard dummy text ever",
    },
    {
      jobtitle: 'Need Cook',
      icon: require('../assets/icons/crown.png'),
      image: require('../images/tutorial.png'),
      detail: "Lorem Ipsum has been the industrys standard dummy text ever",
    }
        ]
}



class ChooseLanguageScreen extends Component {
  constructor(props){
      super(props);
      this.state={
        tabindex : 0
      }
  }
  tabrender(){
    return tabItems.map((value, index)=>{
      return (
        <TouchableOpacity key={index} onPress={()=>this.setState({
            tabindex: index
          })}
          style={{
          // backgroundColor: this.state.tabindex === index ? "blue": "transparent",
          height : scale(40),
          width: "50%",
          justifyContent: "space-between",
          alignItems:'center',
        }}>
        <Text
          style={{
            color: this.state.tabindex === index ? "#fff" : "rgb(158, 212, 247)",
            fontSize: scale(16),
            fontFamily: 'Montserrat-Bold'
          }}>{value}</Text>
        <View style={{
            width: scale(70),
            height: this.state.tabindex === index ? scale(3) : 0,
            backgroundColor: "#fff",
            borderRadius : scale(3)
            // borderColor: this.state.tabindex === index ? "#fff": "transparent"
          }}/>
      </TouchableOpacity>
    )
    })
  }

    render() {
       return (
           <View style={{flex:1}}>
               <LinearGradient
                   colors={['#3E85EF', '#3EBDEF']}
                   start={{x: 0, y: 0}}
                   end={{x: 1, y: 0}}
                   style={{
                       flex: 1
                   }}>
                   <HeaderScreen
                       header={
                           <Header
                               navigation={this.props.navigation}
                               left = {
                                <View style={{backgroundColor: 'transparent', justifyContent: "center", alignItems: 'center', flexDirection:"row"}}>
                                    <TouchableOpacity  onPress={() => this.props.navigation.goBack()}  style={{width: "50%", height:scale(54), backgroundColor: 'transparent', justifyContent: "center", alignItems: 'center'}}>
                                      <Image source={back_arrow} style={{ width: '50%', height: scale(20)}} resizeMode="contain" resizeMethod="resize"/>
                                    </TouchableOpacity>
                                    <TouchableOpacity onPress={() => this.props.navigation.openDrawer()} style={{width: "50%", height:scale(54), backgroundColor: 'transparent', justifyContent: "center", alignItems: 'center'}}>
                                       <Image source={menu} style={{ width: '50%', height: scale(20)}} resizeMode="contain" resizeMethod="resize"/>
                                    </TouchableOpacity>
                                </View>
                               }
                               title={
                                <View style={{ justifyContent : 'center', alignItems: 'flex-start', height:scale(54)}}>
                                   <Text style={{ fontFamily: 'Montserrat-Bold', color:"#fff", fontSize: scale(20)}}>Choose a Language</Text>
                               </View>
                               }
                               right={
                                   <View style={{backgroundColor: 'transparent', justifyContent: "center", alignItems: 'center', flexDirection:"row"}}>
                                     
                                   </View>
                               }
                           />
                       }
                       content={
                         null
                       }
                   />
                   <ScrollView style={{backgroundColor :"rgb(249,252, 255)"}}>
                          <LanguageList/>
                       </ScrollView>
               </LinearGradient>
              
           </View>
       )
  }
}

const styles = StyleSheet.create({
  container: {
    flex:1
  },
     modal: {
    // justifyContent: 'center',
    // alignItems: 'center'
  },

  modal2: {
    height: scale(230),
    backgroundColor: "#3B5998"
  },

  modal3: {
    height: scale(300),
    width: scale(300)
  },
    button:{
    justifyContent:'center',
    alignItems:'center',
    width: scale(100),
    height: scale(40),
    borderRadius:scale(20),
    // borderWidth: 1,
    // borderColor: '#008000',
    paddingTop:scale(5),
    paddingBottom:scale(5),
},
})

export default ChooseLanguageScreen;
