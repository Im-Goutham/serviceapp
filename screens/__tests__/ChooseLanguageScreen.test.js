jest.unmock('redux-mock-store')
jest.unmock('redux-thunk')
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import ChooseLanguageScreen from '../ChooseLanguageScreen';
import {initialState} from '../../config/jest/mockStore';

let store;
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('ChooseLanguageScreen', () => {
    let wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(
            <ChooseLanguageScreen
                store={store}
                // navigation={{state: {params: {mainScreen:''}}}}
                />).dive();
    //    wrapper.setProps({navigation:});
    })
    it('snapshot',()=> {
        expect(wrapper).toMatchSnapshot();
    })
});
