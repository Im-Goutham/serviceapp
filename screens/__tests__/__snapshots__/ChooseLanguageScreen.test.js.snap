// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseLanguageScreen Snapshot testing should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "#3E85EF",
          "#3EBDEF",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      start={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <HeaderScreen
        content={null}
        header={
          <Header
            left={
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 46.95652173913044,
                      "justifyContent": "center",
                      "width": "50%",
                    }
                  }
                >
                  <Image
                    resizeMethod="resize"
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../assets/icons/arrow_left.png",
                      }
                    }
                    style={
                      Object {
                        "height": 17.39130434782609,
                        "width": "50%",
                      }
                    }
                  />
                </TouchableOpacity>
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 46.95652173913044,
                      "justifyContent": "center",
                      "width": "50%",
                    }
                  }
                >
                  <Image
                    resizeMethod="resize"
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../assets/icons/menu.png",
                      }
                    }
                    style={
                      Object {
                        "height": 17.39130434782609,
                        "width": "50%",
                      }
                    }
                  />
                </TouchableOpacity>
              </Component>
            }
            navigation={
              Object {
                "goBack": [MockFunction],
                "navigate": [MockFunction],
              }
            }
            right={
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              />
            }
            title={
              <Component
                style={
                  Object {
                    "alignItems": "flex-start",
                    "height": 46.95652173913044,
                    "justifyContent": "center",
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "color": "#fff",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 17.39130434782609,
                    }
                  }
                >
                  Choose a Language
                </Component>
              </Component>
            }
          />
        }
      />
      <ScrollView
        style={
          Object {
            "backgroundColor": "rgb(249,252, 255)",
          }
        }
      >
        <Connect(LanguageList) />
      </ScrollView>
    </LinearGradient>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LinearGradient
        colors={
          Array [
            "#3E85EF",
            "#3EBDEF",
          ]
        }
        end={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        start={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <HeaderScreen
          content={null}
          header={
            <Header
              left={
                <Component
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                        "width": "50%",
                      }
                    }
                  >
                    <Image
                      resizeMethod="resize"
                      resizeMode="contain"
                      source={
                        Object {
                          "testUri": "../../../assets/icons/arrow_left.png",
                        }
                      }
                      style={
                        Object {
                          "height": 17.39130434782609,
                          "width": "50%",
                        }
                      }
                    />
                  </TouchableOpacity>
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                        "width": "50%",
                      }
                    }
                  >
                    <Image
                      resizeMethod="resize"
                      resizeMode="contain"
                      source={
                        Object {
                          "testUri": "../../../assets/icons/menu.png",
                        }
                      }
                      style={
                        Object {
                          "height": 17.39130434782609,
                          "width": "50%",
                        }
                      }
                    />
                  </TouchableOpacity>
                </Component>
              }
              navigation={
                Object {
                  "goBack": [MockFunction],
                  "navigate": [MockFunction],
                }
              }
              right={
                <Component
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                />
              }
              title={
                <Component
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "height": 46.95652173913044,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Component
                    style={
                      Object {
                        "color": "#fff",
                        "fontFamily": "Montserrat-Bold",
                        "fontSize": 17.39130434782609,
                      }
                    }
                  >
                    Choose a Language
                  </Component>
                </Component>
              }
            />
          }
        />
        <ScrollView
          style={
            Object {
              "backgroundColor": "rgb(249,252, 255)",
            }
          }
        >
          <Connect(LanguageList) />
        </ScrollView>
      </LinearGradient>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <HeaderScreen
            content={null}
            header={
              <Header
                left={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/arrow_left.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/menu.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                  </Component>
                }
                navigation={
                  Object {
                    "goBack": [MockFunction],
                    "navigate": [MockFunction],
                  }
                }
                right={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  />
                }
                title={
                  <Component
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Component
                      style={
                        Object {
                          "color": "#fff",
                          "fontFamily": "Montserrat-Bold",
                          "fontSize": 17.39130434782609,
                        }
                      }
                    >
                      Choose a Language
                    </Component>
                  </Component>
                }
              />
            }
          />,
          <ScrollView
            style={
              Object {
                "backgroundColor": "rgb(249,252, 255)",
              }
            }
          >
            <Connect(LanguageList) />
          </ScrollView>,
        ],
        "colors": Array [
          "#3E85EF",
          "#3EBDEF",
        ],
        "end": Object {
          "x": 1,
          "y": 0,
        },
        "start": Object {
          "x": 0,
          "y": 0,
        },
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": null,
            "header": <Header
              left={
                <Component
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                        "width": "50%",
                      }
                    }
                  >
                    <Image
                      resizeMethod="resize"
                      resizeMode="contain"
                      source={
                        Object {
                          "testUri": "../../../assets/icons/arrow_left.png",
                        }
                      }
                      style={
                        Object {
                          "height": 17.39130434782609,
                          "width": "50%",
                        }
                      }
                    />
                  </TouchableOpacity>
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                        "width": "50%",
                      }
                    }
                  >
                    <Image
                      resizeMethod="resize"
                      resizeMode="contain"
                      source={
                        Object {
                          "testUri": "../../../assets/icons/menu.png",
                        }
                      }
                      style={
                        Object {
                          "height": 17.39130434782609,
                          "width": "50%",
                        }
                      }
                    />
                  </TouchableOpacity>
                </Component>
              }
              navigation={
                Object {
                  "goBack": [MockFunction],
                  "navigate": [MockFunction],
                }
              }
              right={
                <Component
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                />
              }
              title={
                <Component
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "height": 46.95652173913044,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Component
                    style={
                      Object {
                        "color": "#fff",
                        "fontFamily": "Montserrat-Bold",
                        "fontSize": 17.39130434782609,
                      }
                    }
                  >
                    Choose a Language
                  </Component>
                </Component>
              }
            />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(LanguageList) />,
            "style": Object {
              "backgroundColor": "rgb(249,252, 255)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LinearGradient
          colors={
            Array [
              "#3E85EF",
              "#3EBDEF",
            ]
          }
          end={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          start={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <HeaderScreen
            content={null}
            header={
              <Header
                left={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/arrow_left.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/menu.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                  </Component>
                }
                navigation={
                  Object {
                    "goBack": [MockFunction],
                    "navigate": [MockFunction],
                  }
                }
                right={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  />
                }
                title={
                  <Component
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Component
                      style={
                        Object {
                          "color": "#fff",
                          "fontFamily": "Montserrat-Bold",
                          "fontSize": 17.39130434782609,
                        }
                      }
                    >
                      Choose a Language
                    </Component>
                  </Component>
                }
              />
            }
          />
          <ScrollView
            style={
              Object {
                "backgroundColor": "rgb(249,252, 255)",
              }
            }
          >
            <Connect(LanguageList) />
          </ScrollView>
        </LinearGradient>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <HeaderScreen
              content={null}
              header={
                <Header
                  left={
                    <Component
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <TouchableOpacity
                        activeOpacity={0.2}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "height": 46.95652173913044,
                            "justifyContent": "center",
                            "width": "50%",
                          }
                        }
                      >
                        <Image
                          resizeMethod="resize"
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/icons/arrow_left.png",
                            }
                          }
                          style={
                            Object {
                              "height": 17.39130434782609,
                              "width": "50%",
                            }
                          }
                        />
                      </TouchableOpacity>
                      <TouchableOpacity
                        activeOpacity={0.2}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "height": 46.95652173913044,
                            "justifyContent": "center",
                            "width": "50%",
                          }
                        }
                      >
                        <Image
                          resizeMethod="resize"
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/icons/menu.png",
                            }
                          }
                          style={
                            Object {
                              "height": 17.39130434782609,
                              "width": "50%",
                            }
                          }
                        />
                      </TouchableOpacity>
                    </Component>
                  }
                  navigation={
                    Object {
                      "goBack": [MockFunction],
                      "navigate": [MockFunction],
                    }
                  }
                  right={
                    <Component
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        }
                      }
                    />
                  }
                  title={
                    <Component
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Component
                        style={
                          Object {
                            "color": "#fff",
                            "fontFamily": "Montserrat-Bold",
                            "fontSize": 17.39130434782609,
                          }
                        }
                      >
                        Choose a Language
                      </Component>
                    </Component>
                  }
                />
              }
            />,
            <ScrollView
              style={
                Object {
                  "backgroundColor": "rgb(249,252, 255)",
                }
              }
            >
              <Connect(LanguageList) />
            </ScrollView>,
          ],
          "colors": Array [
            "#3E85EF",
            "#3EBDEF",
          ],
          "end": Object {
            "x": 1,
            "y": 0,
          },
          "start": Object {
            "x": 0,
            "y": 0,
          },
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": null,
              "header": <Header
                left={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/arrow_left.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                    <TouchableOpacity
                      activeOpacity={0.2}
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "height": 46.95652173913044,
                          "justifyContent": "center",
                          "width": "50%",
                        }
                      }
                    >
                      <Image
                        resizeMethod="resize"
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/icons/menu.png",
                          }
                        }
                        style={
                          Object {
                            "height": 17.39130434782609,
                            "width": "50%",
                          }
                        }
                      />
                    </TouchableOpacity>
                  </Component>
                }
                navigation={
                  Object {
                    "goBack": [MockFunction],
                    "navigate": [MockFunction],
                  }
                }
                right={
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  />
                }
                title={
                  <Component
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "height": 46.95652173913044,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Component
                      style={
                        Object {
                          "color": "#fff",
                          "fontFamily": "Montserrat-Bold",
                          "fontSize": 17.39130434782609,
                        }
                      }
                    >
                      Choose a Language
                    </Component>
                  </Component>
                }
              />,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(LanguageList) />,
              "style": Object {
                "backgroundColor": "rgb(249,252, 255)",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [MockFunction],
          },
        ],
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
